### PR TITLE
Implement simple MavenIntegrationTest

### PR DESCRIPTION
**Pull request by sellmair**

### CODE REVIEW

File: integration-tests/maven/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    implementation(kotlin("test-junit"))
+}
+
+tasks.integrationTest {
+    dependsOnMavenLocalPublication()
+    dependsOn(":runners:maven-plugin:setupMaven")
+
+    val dokka_version: String by project
+    environment("DOKKA_VERSION", dokka_version)
+    environment("MVN_BINARY_PATH", project(":runners:maven-plugin").extra["MVN_BINARY_PATH"].toString())
```
Comment (sellmair) [inline integration-tests/maven/build.gradle.kts:16]: @kamildoleglo Any ideas? 😨

Comment (sellmair) [reply integration-tests/maven/build.gradle.kts:16]: I will create a proper task called "setupMaven" that will hopefully make this better 👍

File: buildSrc/src/main/kotlin/org/jetbrains/SetupMaven.kt [hunk 2]
```diff
@@ -0,0 +1,49 @@
+package org.jetbrains
+
+import org.gradle.api.DefaultTask
```
Comment (kamildoleglo) [inline buildSrc/src/main/kotlin/org/jetbrains/SetupMaven.kt:3]: Unused import

File: buildSrc/src/main/kotlin/org/jetbrains/taskUtils.kt [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+fun Task.dependsOnMavenLocalPublication() {
+        project.rootProject.allprojects.forEach { otherProject ->
+            otherProject.invokeWhenEvaluated { evaluatedProject ->
+                evaluatedProject.tasks.findByName("publishToMavenLocal")?.let { publishingTask ->
+                    this.dependsOn(publishingTask)
+                }
+            }
+        }
+}
```
Comment (kamildoleglo) [inline buildSrc/src/main/kotlin/org/jetbrains/taskUtils.kt:13]: Indented one level too much

File: integration-tests/build.gradle.kts [hunk 4]
```diff
@@ -37,4 +37,5 @@ subprojects {
 dependencies {
     implementation(kotlin("stdlib"))
     implementation(kotlin("test-junit"))
-}
+    val coroutines_version: String by project
+    api("org.jetbrains.kotlinx:kotlinx-coroutines-core:$coroutines_version")}
```
Comment (kamildoleglo) [inline integration-tests/build.gradle.kts:41]: Why do we need `api` here? Wouldn't `implementation` be sufficient?

File: integration-tests/cli/build.gradle.kts [hunk 5]
```diff
@@ -35,6 +31,7 @@ val basePluginShadowJar by tasks.register("basePluginShadowJar", ShadowJar::clas
 }
 
 tasks.integrationTest {
+    inputs.dir((file("projects")))
```
Comment (kamildoleglo) [inline integration-tests/cli/build.gradle.kts:34]: One pair of brackets too much

File: buildSrc/src/main/kotlin/org/jetbrains/SetupMaven.kt (hunk 2 above)
Comment (sellmair) [reply buildSrc/src/main/kotlin/org/jetbrains/SetupMaven.kt:3]: Removed ✅

File: buildSrc/src/main/kotlin/org/jetbrains/taskUtils.kt (hunk 3 above)
Comment (sellmair) [reply buildSrc/src/main/kotlin/org/jetbrains/taskUtils.kt:13]: Fixed ✅

File: integration-tests/build.gradle.kts (hunk 4 above)
Comment (sellmair) [reply integration-tests/build.gradle.kts:41]: Yes, you are right. I for some reason thought that I am exposing some coroutines stuff in the API, but nope!
Changed to `implementation` ✅

File: integration-tests/cli/build.gradle.kts (hunk 5 above)
Comment (sellmair) [reply integration-tests/cli/build.gradle.kts:34]: Thanks! Changed ✅

**Pull request merged**